docs/2-Defining_parameters.md

```diff
@@ -11,24 +11,29 @@ be mixed (i.e., do not nest a Settings section in a Config section and vice vers
 
 ## Defining non-container section(s)
 
-A section is defined by subclassing the relevant base class (ConfigSectionBase for
-config, SettingsSectionBase for settings) and decorating it with
+A section is defined by subclassing the relevant base class (`ConfigSectionBase` for
+config, `SettingsSectionBase` for settings) and decorating it with
 `@dataclass(frozen=True)`. Parameters are defined as fields of the dataclass. For a 
 dataclass, it is mandatory to add a type hint for each field. These type hints are used
-also to validate the data that is read from the parameter file (because we are
-using [pydantic dataclasses](https://docs.pydantic.dev/usage/dataclasses/)). If you
+also to validate the data that is read from the parameter file. If you
 specify a default value as well, then you prevent the occurance of an exception if the
 value for the parameter of concern is not found in the parameter file.
 
+The `dataclass` decorator provided by `application_settings` is actually an exported
+[`pydantic.dataclasses.dataclass`](https://docs.pydantic.dev/usage/dataclasses/), which
+is a drop-in replacement for a `dataclass` from the standard python library, enhanced
+with data validation. If you need additional dataclass functionality such as `fields`
+etc., you can import those from the standard library.
+
 Nested sections are obtained by defining fields in a section that are type hinted with 
 the appropriate contained section class(es) and instantiated (possible only when all
 parameters of the nested section have default values).
 
 ## Defining the container
 
 The container is a special section that is to be the root for parametrisation of an
-application. It is defined likewise: by subclassing the relevant base class (ConfigBase
-for config, SettingsBase for settings), decorating it with `@dataclass(frozen=True)`,
+application. It is defined likewise: by subclassing the relevant base class (`ConfigBase`
+for config, `SettingsBase` for settings), decorating it with `@dataclass(frozen=True)`,
```

Comment:
What happens if we don't decorate it? And should we add an issue to add checking to the class that this is actually done?

Reply:
If you don't decorate a container/section that has a subsection, then you will get a printed message that you access a section that has not been loaded. If you don't decorate a leaf section, then you will not get notified. And data loading will not work.
I really don't know how to test if the decorator is there. There is a test 'is_dataclass' but that will always return true because the superclass is a dataclass. So I see no use for creating an issue.

The best we can do perhaps is to add a note to the docs what happens if one forgets the decorator. Do you think that is worthwile?